pennylane/capture/autograph/ag_primitives.py

```diff
@@ -84,6 +94,90 @@ def functional_cond():
     set_state(results)
 
 
+def assert_iteration_inputs(inputs, symbol_names):
+    """All loop carried values, variables that are updated each iteration or accessed after the
+    loop terminates, need to be initialized prior to entering the loop.
+
+    The reason is two-fold:
+      - the type information from those variables is required for tracing
+      - we want to avoid access of a variable that is uninitialized, or uninitialized in a subset
```

Comment:
```suggestion
      - we want to avoid access to a variable that is uninitialized, or uninitialized in a subset
```

If I understand correctly, with this you mean that you want to avoid accessing such a variable inside the loop (as in `test_uninitialized_variables`).

Reply:
Yes, that's right - do you think it comes across clearly? I've now changed this to "we want to avoid accessing a variable that is..."